src/nyc_trees/apps/home/training/types.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+from django.core.urlresolvers import reverse
+from django.contrib.auth.decorators import login_required
+from django_tinsel.decorators import route
+from django_tinsel.utils import decorate as do
+from apps.home.training.decorators import render_flatpage, mark_user
+from django.contrib.flatpages.views import flatpage
+
+_PURE_TEMPLATE = '%s_pure'
+_MARK_PROGRESS_TEMPLATE = '%s_mark_progress'
+_TRAINING_FINISHED_TEMPLATE = 'training_finished_%s'
+
+
+class TrainingGateway(object):
+    def __init__(self, view, first_step, last_step):
+        self.previous_step = last_step
+        self.next_step = first_step
+        self.view = view
+        self.name = None
+
+    def is_flatpage(self):
+        return False
```

Comment:
No, not really. They both implement the same interface through ducktyping, but don't share code or data so inheritance is not useful.  It's hard to convey that in how they are defined, because interfaces and abstract classes are not a thing in python.